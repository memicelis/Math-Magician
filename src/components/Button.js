import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, onClick }) => {
  const handleClick = () => {
    onClick(value);
  };
  const buttonClass = value === '0'
    ? 'calculator-button calculator-button--0'
    : 'calculator-button';
  return (
    <button className={buttonClass} type="button" onClick={handleClick}>
      {value}
    </button>
  );
};

Button.propTypes = {
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
