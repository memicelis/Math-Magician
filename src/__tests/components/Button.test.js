import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

it('Button renders correctly', () => {
  const tree = renderer
    .create(<Button value="5" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button renders correctly when value is 0', () => {
  const tree = renderer
    .create(<Button value="0" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
