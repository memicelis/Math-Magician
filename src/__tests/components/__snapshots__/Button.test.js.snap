// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="calculator-button"
  onClick={[Function]}
  type="button"
>
  5
</button>
`;

exports[`Button renders correctly when value is 0 1`] = `
<button
  className="calculator-button calculator-button--0"
  onClick={[Function]}
  type="button"
>
  0
</button>
`;
